tests/vm.rs

```diff
@@ -557,10 +559,14 @@ fn run_expr_int() {
 
     let text = r#"io.run_expr "123" "#;
     let mut vm = make_vm();
-    let (result, _) = Compiler::new().run_io_expr::<String>(&mut vm, "<top>", text).unwrap();
-    let expected = "123 : Int";
-
-    assert_eq!(result, expected);
+    let (result, _) = Compiler::new().run_io_expr::<IO<String>>(&mut vm, "<top>", text).unwrap();
```

Comment:
Is there a reason why we need to do that kind of internal dispatch as opposed to just having two functions?